libraries/aws/shadow/src/aws_iot_shadow_static_memory.c

```diff
@@ -73,7 +73,7 @@
  * the constant `AWS_IOT_MAX_THING_NAME_LENGTH` is used for the length of
  * #_shadowSubscription_t.pThingName.
  */
-#define SHADOW_SUBSCRIPTION_SIZE    ( sizeof( _shadowSubscription_t ) + AWS_IOT_MAX_THING_NAME_LENGTH )
+#define SHADOW_SUBSCRIPTION_SIZE   ( ( size_t ) ( sizeof( _shadowSubscription_t ) + ( size_t ) AWS_IOT_MAX_THING_NAME_LENGTH ) )
```

Comment:
Would just ` ( size_t ) ( sizeof( _shadowSubscription_t ) +AWS_IOT_MAX_THING_NAME_LENGTH ) ` not suffice?

Reply:
The issue is that the two sides of the operands differed in signedness.

Reply:
Do we still need the size_t case on the whole expression?

Reply:
No.